test/integration/replicaset/replicaset_test.go

```diff
@@ -279,39 +285,48 @@ func TestAdoption(t *testing.T) {
 		close(stopCh)
 	}
 }
-
-func createRSsPods(t *testing.T, clientSet clientset.Interface, rss []*v1beta1.ReplicaSet, pods []*v1.Pod, ns string) {
+*/
+func createRSsPods(t *testing.T, clientSet clientset.Interface, rss []*v1beta1.ReplicaSet, pods []*v1.Pod, ns string) ([]*v1beta1.ReplicaSet, []*v1.Pod) {
 	rsClient := clientSet.Extensions().ReplicaSets(ns)
 	podClient := clientSet.Core().Pods(ns)
+	var createdRSs []*v1beta1.ReplicaSet
+	var createdPods []*v1.Pod
 	for _, rs := range rss {
-		if _, err := rsClient.Create(rs); err != nil {
+		createdRS, err := rsClient.Create(rs)
+		if err != nil {
 			t.Fatalf("Failed to create replica set %s: %v", rs.Name, err)
 		}
+		createdRSs = append(createdRSs, createdRS)
 	}
 	for _, pod := range pods {
-		if _, err := podClient.Create(pod); err != nil {
+		createdPod, err := podClient.Create(pod)
+		if err != nil {
 			t.Fatalf("Failed to create pod %s: %v", pod.Name, err)
 		}
+		createdPods = append(createdPods, createdPod)
 	}
+
+	return createdRSs, createdPods
 }
 
 func waitRSStable(t *testing.T, clientSet clientset.Interface, rs *v1beta1.ReplicaSet, ns string) {
 	rsClient := clientSet.Extensions().ReplicaSets(ns)
-	if err := wait.Poll(10*time.Second, 60*time.Second, func() (bool, error) {
-		updatedRS, err := rsClient.Get(rs.Name, metav1.GetOptions{})
+	if err := wait.Poll(10*time.Second, 15*time.Second, func() (bool, error) {
+		rs, err := rsClient.Get(rs.Name, metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}
-		if updatedRS.Status.Replicas != *rs.Spec.Replicas {
+		if rs.Status.Replicas != *rs.Spec.Replicas {
 			return false, nil
 		} else {
 			return true, nil
 		}
 	}); err != nil {
-		t.Fatal(err)
+		t.Fatal(err, rs.Status.Replicas, *rs.Spec.Replicas)
 	}
 }
 
+/*
 func TestUpdateSelectorToAdopt(t *testing.T) {
```

Comment:
@enisoc I think we can remove this test and `TestUpdateSelectorToRemoveControllerRef` test.

Reply:
Yes, your tests can replace those.